crates/worktree/src/worktree.rs

```diff
@@ -2658,14 +2658,24 @@ impl Snapshot {
     }
 
     pub fn child_entries<'a>(&'a self, parent_path: &'a Path) -> ChildEntriesIter<'a> {
+        self.child_entries_with_options(parent_path, true, true, true)
+    }
+
+    pub fn child_entries_with_options<'a>(
+        &'a self,
+        parent_path: &'a Path,
+        include_files: bool,
```

Comment:
Could we introduce a struct to contain these options?

Having a signature with three consecutive booleans like this really harms readability at the callsite.